src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs

```diff
@@ -6,12 +6,34 @@
 using System.Diagnostics;
 using System.Linq;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Core;
 using Microsoft.Extensions.DependencyModel;
+using Microsoft.Extensions.PlatformAbstractions;
 
 namespace Microsoft.AspNetCore.Mvc.Infrastructure
 {
+    /// <summary>
+    /// An <see cref="IAssemblyProvider"/> that uses <see cref="DependencyContext"/> to discover assemblies that may
+    /// contain Mvc specific types such as controllers, and view components.
+    /// </summary>
     public class DependencyContextAssemblyProvider : IAssemblyProvider
     {
+        private readonly DependencyContext _dependencyContext;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DependencyContextAssemblyProvider"/>.
+        /// </summary>
+        /// <param name="environment">The <see cref="IApplicationEnvironment"/>.</param>
+        public DependencyContextAssemblyProvider(IApplicationEnvironment environment)
+        {
+#if NET451
+            var applicationAssembly = Assembly.Load(environment.ApplicationName);
+#else
+            var applicationAssembly = Assembly.Load(new AssemblyName(environment.ApplicationName));
+#endif
+            _dependencyContext = DependencyContext.Load(applicationAssembly);
```

Comment:
I guess we can change default logic to use `AssemblyName(environment.ApplicationName)` by default in dependency context. It default to same anyway.